### ISSUE TITLE
NPE ActionBar.TabListener

### ISSUE THREAD

**Issue by gabrielittner** [JakeWharton/ActionBarSherlock#391]
I want to use the Tabs to Navigate between two fragments and always get a NullPointer Exception at startup. This is my code (which is from here http://developer.android.com/guide/topics/ui/actionbar.html#Tabs):

```
public static class TabListener<T extends Fragment> implements ActionBar.TabListener {
    private Fragment mFragment;
    private final Activity mActivity;
    private final String mTag;
    private final Class<T> mClass;

    public TabListener(Activity activity, String tag, Class<T> clz) {
        mActivity = activity;
        mTag = tag;
        mClass = clz;
    }

    public void onTabSelected(Tab tab, FragmentTransaction ft) {
        if (mFragment == null) {
            mFragment = Fragment.instantiate(mActivity, mClass.getName());
            ft.add(R.id.main_content, mFragment);
        } else {
            ft.attach(mFragment);
        }
    }

    public void onTabUnselected(Tab tab, FragmentTransaction ft) {
        if (mFragment != null) {
            ft.detach(mFragment);
        }
    }

    public void onTabReselected(Tab tab, FragmentTransaction ft) {
    }
}
```

The NullPointerException is thrown here: 

```
    ft.add(R.id.main_content, mFragment);
```

The fragment isn't null so it has to be the layout. Changing it to android.R.id.content also throws the exception. All my imports are right (from support-v4 and abs). There is probably a bug in the implementation of ActionBar.TabListener and it's FragmentTransactions, doing the getSupportFragmentManager().beginTransaction() solves it.

```
public static class TabListener<T extends Fragment> implements ActionBar.TabListener {
    private Fragment mFragment;
    private final Activity mActivity;
    private final String mTag;
    private final Class<T> mClass;

    public TabListener(Activity activity, String tag, Class<T> clz) {
        mActivity = activity;
        mTag = tag;
        mClass = clz;
    }

    public void onTabSelected(Tab tab, FragmentTransaction ft) {
        if (mFragment == null) {
            mFragment = Fragment.instantiate(mActivity, mClass.getName());
            ((FragmentActivity) mActivity).getSupportFragmentManager().beginTransaction().add(R.id.main_content, mFragment, mTag).commit();
        } else {
             ((FragmentActivity) mActivity).getSupportFragmentManager().beginTransaction().attach(mFragment).commit();
        }
    }

    public void onTabUnselected(Tab tab, FragmentTransaction ft) {
        if (mFragment != null) {
             ((FragmentActivity) mActivity).getSupportFragmentManager().beginTransaction().detach(mFragment).commit();
        }
    }

    public void onTabReselected(Tab tab, FragmentTransaction ft) {
    }
}
```

I will try to find time for fixing and commiting it myself in the next days.

**Comment by JakeWharton** [comment]
Does this happen on ICS, pre-ICS, or both?

**Comment by vRallev** [comment]
Happens on both. I had the same issue. The FragmentTransaction is null. So I had to check first whether it is null or not and if so, I've created my own transaction and committed it manually.

**Comment by ericklee** [comment]
I have this issue on ICS too but not on pre-ICS.

**Comment by gabrielittner** [comment]
I fixed it. Explanation and commit coming in the next hour.

Although I don't know @vRallev had the same issue, because it only occures on ICS for me.

**Comment by vRallev** [comment]
I'm sorry for the confusion, I've checked it again and the emulator (x86 2.3.3) didn't have this bug now. I think your are right gabrielittner.

**Comment by gabrielittner** [comment]
Is it possible to add the pull request to this issue or do I have to simply send a pull request which opens a new issue and reference this one here?

**Comment by JakeWharton** [comment]
You can only convert an issue into a pull request from the API. Just create
a new one and reference this issue number in the body.

---

Jake Wharton
http://about.me/jakewharton

On Fri, Mar 23, 2012 at 9:17 AM, gabrielittner <
reply@reply.github.com

**Comment by JakeWharton** [comment]
Fixed with #402.
